scalding-spark/src/main/scala/com/twitter/scalding/spark_backend/Op.scala

```diff
@@ -142,14 +142,14 @@ object Op extends Serializable {
 
     def run(session: SparkSession)(implicit ec: ExecutionContext): Future[RDD[_ <: A]] =
       cache.getOrElseUpdate(session,
-        input.run(session).map { rdd => fn(widen(rdd)) })
+        CFuture(input.run(session).map { rdd => fn(widen(rdd)) }, CancellationHandler.empty)).future
```

Comment:
I think we should have two FutureCache implementations or use a typeclass approach to make FutureCache generic. Copying FutureCache to CFutureCache and not changing the original might work here.

Reply:
ah, the typeclass makes things nicer, thanks